File: LeetCode/Easy/3151-special-array-i/3151-special-array-i.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        is_odd = True
        
        if not nums[0] % 2:
            is_odd = False
        
        for i in range(1, len(nums)):
            if is_odd == nums[i] % 2:
                return False
            is_odd = not is_odd
        
        return True

File: LeetCode/Easy/3151-special-array-i/3151-special-array-i.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        bool isOdd = true;
        
        if (nums[0] % 2 == 0) isOdd = false;
        
        for (int i = 1; i < nums.size(); ++i){
            if (nums[i] % 2 == isOdd) return false;
            
            isOdd = !isOdd;
        }
        
        return true;
    }
};